src/quaternion.rs

```diff
@@ -126,30 +126,31 @@ impl<S: BaseFloat> Quaternion<S> {
     ///   (http://number-none.com/product/Understanding%20Slerp,%20Then%20Not%20Using%20It/)
     /// - [Arcsynthesis OpenGL tutorial]
     ///   (http://www.arcsynthesis.org/gltut/Positioning/Tut08%20Interpolation.html)
-    pub fn slerp(self, other: Quaternion<S>, amount: S) -> Quaternion<S> {
-        let dot = self.dot(other);
-        let dot_threshold = cast(0.9995f64).unwrap();
+    pub fn slerp(self, mut other: Quaternion<S>, mut amount: S) -> Quaternion<S> {
+        let mut dot = self.dot(other);
+        let dot_threshold: S = cast(0.9995f64).unwrap();
+
+        // Fix for opposite quaternions: pick a perpendicular target and change the amount accordingly
+        // Therefore, the following code cannot assume that 0 <= amount <= 1.
+        if ulps_eq!(dot, -S::one()) {
```

Comment:
why is -1 compared by `ulps_eq` but 1 is compared by the `dot_threshold` thing? Can we align these?

Reply:
1 is compared by `dot_threshold` because in that range, `slerp` is close enough to `nlerp` to not matter. But in the case of -1, this is not the case, and so we should have a tighter threshold. If the vectors are very close to being on a line but the dot product isn't quite -1, their span is still a plane and `slerp` can still give a good result.

Reply:
Hmm. A quaternion is the same if you multiply all the components by -1. So if the dot becomes negative, you may as well do that operation on one of the operands, and you get a positive dot product. Can we unify therefore the check for proximity to 1 and -1 ?

Reply:
No, because a negative dot product represents a rotation of more than 180 degrees, and `slerp` should respect that.

Reply:
Ok, I'm genuinely confused now. When interpolating between any 2 directions, and choosing the shortest arc, there shouldn't be a rotation exceeding 180 degrees.

Reply:
That's not what `slerp` does, though. It takes the shortest arc only if the dot product is nonnegative. There could be a `shortest_slerp` function that always takes the shortest arc.

Reply:
Interesting. So looking at the example code on the [wiki](https://en.wikipedia.org/wiki/Slerp), they appear to be doing precisely what I'm asking for:
```cpp
    // If the dot product is negative, slerp won't take
    // the shorter path. Note that v1 and -v1 are equivalent when
    // the negation is applied to all four components. Fix by 
    // reversing one quaternion.
    if (dot < 0.0f) {
        v1 = -v1;
        dot = -dot;
    }
```

Reply:
Wouldn't that need to be done for `nlerp` as well for consistency? And I notice that in [this post](http://number-none.com/product/Understanding%20Slerp,%20Then%20Not%20Using%20It/), the author doesn't do that reversing.

Reply:
good question! it sounds like we should do that for nlerp as well

Reply:
I made them both take the shortest path now.